### PR TITLE
Multicall multicontract

### PR DESCRIPTION
**Pull request by gusjavaz**
## Scope
This PR adds two new hardhat tasks:

1. generateContractsDescriptor: Generates a JSON descriptor file with all contract deployment information requirements using (slow) function getBuildInfo (requires deploymentList and deploymentArgsTemplate files to be already generated) . 
2. deployContractsFromDescriptor: Deploy contracts faster with information from previously created JSON file.

## Why?
To speed up contract deployment through deployContracts current task, tests shows deployContractsFromDescriptor is 5 times faster against hardhat node and almost 2 times faster against geth.

### CODE REVIEW

File: bridge/scripts/lib/deployment/deploymentUtil.ts [hunk 1]
```diff
@@ -292,3 +326,165 @@ export async function getDeployGroupIndex(
...
+      const deployType = contract.deployType;
+      switch (deployType) {
+        case STATIC_DEPLOYMENT: {
+          let multiCallArgsArray = Array();
+          let [deployTemplate, deployStatic] =
+            await getDeployStaticMultiCallArgs(
+              contract,
+              hre,
+              factoryBase,
+              factory,
```
Comment (z-j-lin) [inline bridge/scripts/lib/deployment/deploymentUtil.ts:439]: can get factory address with factory.address

**Review by z-j-lin — CHANGES_REQUESTED** [review]
dont forget to run npm run lint -- --fix

File: bridge/scripts/lib/deployment/deploymentUtil.ts [hunk 2]
```diff
@@ -1,8 +1,32 @@
...
+import { BytesLike, ContractFactory } from "ethers";
+import {
+  Artifacts,
+  HardhatRuntimeEnvironment,
+  RunTaskFunction,
+} from "hardhat/types";
+import {
+  AliceNetFactory,
+  AliceNetFactory__factory,
+} from "../../../typechain-types";
```
[the remark this answers is not in the document]
Comment (gusjavaz) [reply bridge/scripts/lib/deployment/deploymentUtil.ts]: Resolved as per ZJ's commit

File: bridge/scripts/lib/deployment/deploymentUtil.ts [hunk 3]
```diff
@@ -292,3 +326,165 @@ export async function getDeployGroupIndex(
 ) {
   return await getCustomNSTag(fullName, "deploy-group-index", artifacts);
 }
+
+export async function getDeployStaticMultiCallArgs(
+  contractDescriptor: ContractDescriptor,
+  hre: HardhatRuntimeEnvironment,
+  factoryBase: AliceNetFactory__factory,
+  factory: AliceNetFactory,
```
[the remark this answers is not in the document]
Comment (gusjavaz) [reply bridge/scripts/lib/deployment/deploymentUtil.ts]: Resolved as per ZJ's commit

File: bridge/scripts/lib/deployment/deploymentUtil.ts [hunk 4]
```diff
@@ -292,3 +326,165 @@ export async function getDeployGroupIndex(
...
+  txCount: number
+) {
+  const logicContract: ContractFactory = await hre.ethers.getContractFactory(
```
[the remark this answers is not in the document]
Comment (gusjavaz) [reply bridge/scripts/lib/deployment/deploymentUtil.ts:339]: Resolved as per ZJ's commit

File: bridge/scripts/lib/deployment/deploymentUtil.ts (hunk 1 above)
Comment (gusjavaz) [reply bridge/scripts/lib/deployment/deploymentUtil.ts]: Resolved as per ZJ's commit

**Review by z-j-lin — APPROVED** [review]
all good to go

**Pull request merged**